gap/semigroups/semieunit.gi

```diff
@@ -0,0 +1,715 @@
+  fi;
+
+  fam := NewFamily("McAlisterTripleSemigroupFamily",
+                   IsMcAlisterTripleSemigroupElement);
+
+  # Check if this McAlister triple semigroup is a monoid
+  if IsMeetSemilatticeDigraph(Y) then
+    filt := IsMcAlisterTripleSemigroup and IsMonoid;
+  else
+    filt := IsMcAlisterTripleSemigroup;
+  fi;
+  # Create the semigroup itself
+  M := Objectify(NewType(CollectionsFamily(fam), filt and IsAttributeStoringRep
+                         and IsEUnitaryInverseSemigroup and IsWholeFamily),
+                 rec());
+
+  M!.elementType := NewType(fam, IsMcAlisterTripleSemigroupElementRep);
+
+  SetMcAlisterTripleSemigroupGroup(M, G);
+  SetMcAlisterTripleSemigroupAction(M, anti_act);
+  SetMcAlisterTripleSemigroupPartialOrder(M, X);
+  SetMcAlisterTripleSemigroupSemilattice(M, Y);
+
+  GeneratorsOfSemigroup(M);
+  return M;
+end);
+
+InstallMethod(McAlisterTripleSemigroup,
+"for a perm group, digraph, and digraph",
+[IsPermGroup, IsDigraph, IsDigraph],
+function(G, X, Y)
+  return McAlisterTripleSemigroup(G, OnPoints, X, Y);
+end);
+
+InstallMethod(McAlisterTripleSemigroup,
+"for a perm group, action, digraph, and a homogeneous list",
+[IsGroup, IsFunction, IsDigraph, IsHomogeneousList],
+function(G, act, X, sub_ver)
+  return McAlisterTripleSemigroup(G, act, X, InducedSubdigraph(X, sub_ver));
+end);
+
+InstallMethod(McAlisterTripleSemigroup,
+"for a perm group, digraph, and a homogeneous list",
+[IsPermGroup, IsDigraph, IsHomogeneousList],
+function(G, X, sub_ver)
+  return McAlisterTripleSemigroup(G, OnPoints, X,
+                                  InducedSubdigraph(X, sub_ver));
+end);
+
+#############################################################################
+# Methods for McAlister triple semigroups
+#############################################################################
+InstallMethod(\=, "for two McAlister triple semigroups",
+[IsMcAlisterTripleSemigroup, IsMcAlisterTripleSemigroup],
+function(x, y)
+  local A, G, X, Y, g;
+  G := McAlisterTripleSemigroupGroup(x);
+  X := McAlisterTripleSemigroupPartialOrder(x);
+  Y := McAlisterTripleSemigroupSemilattice(x);
+  if G = McAlisterTripleSemigroupGroup(y)
+      and X = McAlisterTripleSemigroupPartialOrder(y)
+      and Y = McAlisterTripleSemigroupSemilattice(y) then
+    if ForAll(G, g -> ForAll(DigraphVertices(X),
+         A -> McAlisterTripleSemigroupAction(x)(A, g) =
+              McAlisterTripleSemigroupAction(y)(A, g))) then
+      return true;
+    fi;
+  fi;
+  return false;
+end);
+
+InstallMethod(OneImmutable, "for a McAlister triple semigroup",
+[IsMcAlisterTripleSemigroup],
+function(S)
+  local Y;
+  Y := McAlisterTripleSemigroupSemilattice(S);
+  return MTE(S, DigraphSources(DigraphRemoveLoops(Y))[1], ());
+end);
+
+# (A, g) in S if and only if Ag^-1 is a vertex of the semilattice of S
+InstallMethod(AsList, "for a McAlister triple semigroup",
+[IsMcAlisterTripleSemigroup],
+function(S)
+  local out, g, A, V;
+    out := [];
```

Comment:
The line containing `out := [];` should be 2 characters to the left, lined up with `local`. And same for the lines below it.